monitoring/api/v3/cloud-client/snippets.py

```diff
@@ -13,126 +13,215 @@
+    aggregation = monitoring_v3.types.Aggregation()
+    aggregation.alignment_period.seconds = 300  # 5 minutes
+    aggregation.per_series_aligner = (
+        monitoring_v3.enums.Aggregation.Aligner.ALIGN_MEAN)
+    aggregation.cross_series_reducer = (
+        monitoring_v3.enums.Aggregation.Reducer.REDUCE_MEAN)
+    aggregation.group_by_fields.append('resource.zone')
+
+    results = client.list_time_series(
+        project_name,
+        'metric.type = "compute.googleapis.com/instance/cpu/utilization"',
+        interval,
+        monitoring_v3.enums.ListTimeSeriesRequest.TimeSeriesView.FULL,
+        aggregation)
+    for result in results:
         print(result)
     # [END monitoring_read_timeseries_reduce]
 
 
-def list_metric_descriptors():
+def list_metric_descriptors(project_id):
     # [START monitoring_list_descriptors]
-    client = monitoring.Client()
-    for descriptor in client.list_metric_descriptors():
+    client = monitoring_v3.MetricServiceClient()
+    project_name = client.project_path(project_id)
+    for descriptor in client.list_metric_descriptors(project_name):
         print(descriptor.type)
     # [END monitoring_list_descriptors]
 
 
-def list_monitored_resources():
+def list_monitored_resources(project_id):
     # [START monitoring_list_resources]
-    client = monitoring.Client()
-    for descriptor in client.list_resource_descriptors():
+    client = monitoring_v3.MetricServiceClient()
+    project_name = client.project_path(project_id)
+    resource_descriptors = (
+        client.list_monitored_resource_descriptors(project_name))
+    for descriptor in resource_descriptors:
         print(descriptor.type)
     # [END monitoring_list_resources]
 
 
-def get_monitored_resource_descriptor(resource_type_name):
+def get_monitored_resource_descriptor(project_id, resource_type_name):
     # [START monitoring_get_resource]
-    client = monitoring.Client()
-    print(client.fetch_resource_descriptor(resource_type_name))
+    client = monitoring_v3.MetricServiceClient()
+    resource_path = client.monitored_resource_descriptor_path(
+        project_id, resource_type_name)
+    pprint.pprint(client.get_monitored_resource_descriptor(resource_path))
     # [END monitoring_get_resource]
 
 
-def get_metric_descriptor(metric_type_name):
+def get_metric_descriptor(metric_name):
     # [START monitoring_get_descriptor]
-    client = monitoring.Client()
-    print(client.fetch_metric_descriptor(metric_type_name))
+    client = monitoring_v3.MetricServiceClient()
+    descriptor = client.get_metric_descriptor(metric_name)
+    pprint.pprint(descriptor)
     # [END monitoring_get_descriptor]
 
 
+class MissingProjectIdError(Exception):
+    pass
+
+
+def project_id():
+    """Retreives the project id from the environment variable.
+
+    Raises:
+        MissingProjectIdError -- When not set.
+
+    Returns:
+        str -- the project name
+    """
+    project_id = os.environ['GCLOUD_PROJECT']
```

Comment:
Could we check the `GOOGLE_CLOUD_PROJECT` env var first and use `GCLOUD_PROJECT` as a fallback?

Reply:
I haven't seen that in any of the python samples I've come across.  Is that a new standard?